src/_pytest/assertion/util.py

```diff
@@ -169,23 +190,56 @@ def assertrepr_compare(config, op: str, left: Any, right: Any) -> Optional[List[
 
 
 def _compare_eq_any(left: Any, right: Any, verbose: int = 0) -> List[str]:
-    explanation = []  # type: List[str]
+    """
+    Compare 2 objects.
+    The generic comparison function, which calls other functions based on the input's type.
+
+    :param left
+    :type left: Any
+    :param right
+    :type right: Any
+    :param verbose
+    :type verbose: int
+    .. note:: indendation used for better visual perception of the comparison report.
+    :return: explanation of an AssertionError
+    :rtype: List[str]
+    """
+    explanation = []
+
+    my_name = cast(types.FrameType, inspect.currentframe()).f_code.co_name
+    num_calls = _get_number_of_calls(func_name=my_name)  # type: int
```

Comment:
I see why you are using `_get_number_of_calls` here, but couldn't we instead receive a `indentation` parameter as argument, and forward that down the stack like it was done with the other methods? Seems less "magic" than fiddling with frame objects.

Reply:
Also it is probably easier to treat indentation as a single number that increases by one for each indentation level, and then each function that receives the indentation level just calls an utility function to build the indent string:

```python
def _compare_eq_verbose(left: Any, right: Any, indent_level: int) -> List[str]:
    indent_level += 1
    indent = _get_indent(indent_level)

...

def _get_indent(indent_level):
    return " " * (indent_level * 4)
```

Reply:
Thank you @nicoddemus for reviewing the changes! 
I am not sure that I understand your proposal right: you propose to pass the `indentation` (or `indent_level`) as argument to `_compare_eq_any()`, right? But how would this function (or the calling function - `assertrepr_compare`) know, how many times was it called (and it needs that to make the right indentation, dependent on the level of nestedness of data)? We could make a wrapper around the function, which would store the state (number of executions), but that would be reinvention of the call stack. Or do I get your idea completely wrong?

Regarding having `_get_indent()` - you propose to call this function directly from every function, that needs indentation, right? Wouldn't it create additional overhead? An option would be to call this function just from `_compare_eq_any()`, instead of creating the indent string there, and pass the string as parametere, as I do now. But when you think, that overhead is fine, and from design perspective it's better to call the `_get_indent()` directly from every needed function, I'll do that.

Reply:
I think, I got some clarification from @bluetech

Reply:
Thanks! I'm assuming you got my suggestion then, but let me know otherwise and I will try to explain it.